features/nationaliteithistorie/dev/nationaliteit-gegevens.feature

```diff
@@ -0,0 +1,129 @@
+      | type                | RaadpleegMetPeildatum                                                                                     |
+      | burgerservicenummer | 000009830                                                                                                 |
+      | peildatum           | 2022-08-16                                                                                                |
+      | fields              | nationaliteiten.nationaliteit.code,nationaliteiten.redenOpname.code,nationaliteiten.datumIngangGeldigheid |
+      Dan heeft de response de volgende 'nationaliteiten'
+      | type          | nationaliteit.code | redenOpname.code | datumIngangGeldigheid.type | datumIngangGeldigheid.datum | datumIngangGeldigheid.langFormaat |
+      | Nationaliteit | 0001               | 001              | Datum                      | 1975-07-07                  | 7 juli 1975                       |
+
+
+  Rule: Voor een beëindigde nationaliteit of beëindigd bijzonder Nederlanderschap worden type, nationaliteit, aanduidingBijzonderNederlanderschap, redenOpname en datumIngangGeldigheid overgenomen uit de jongste bijbehorende historische categorie (54) waarin deze voorkomen en die niet onjuist is.
+    
+    Scenario: persoon heeft een beëindigde nationaliteit
+      Gegeven de persoon met burgerservicenummer '999993008' heeft een 'nationaliteit' met de volgende gegevens
+      | nationaliteit (05.10) | reden opnemen (63.10) | reden beëindigen (64.10) | datum ingang geldigheid (85.10) |
+      |                       |                       | 404                      | 20050131                        |
+      | 0131                  | 301                   |                          | 19750501                        |
+      Als nationaliteithistorie wordt geraadpleegd met de volgende parameters
+      | naam                | waarde                                                                                                    |
+      | type                | RaadpleegMetPeriode                                                                                       |
+      | burgerservicenummer | 999993008                                                                                                 |
+      | peildatum           | 2000-01-01                                                                                                |
+      | fields              | nationaliteiten.nationaliteit.code,nationaliteiten.redenOpname.code,nationaliteiten.datumIngangGeldigheid |
+      Dan heeft de response de volgende 'nationaliteiten'
+      | type          | nationaliteit.code | redenOpname.code | datumIngangGeldigheid.type | datumIngangGeldigheid.datum | datumIngangGeldigheid.langFormaat |
+      | Nationaliteit | 0131               | 301              | Datum                      | 1975-05-01                  | 1 mei 1975                        |
```

Comment:
Wat vinden jullie ervan om deze scenario als volgt te schrijven?

    Scenario: verlies bijzonder Nederlanderschap
      Gegeven de persoon met burgerservicenummer '555550001' heeft een 'nationaliteit' met de volgende gegevens
      | bijzonder Nederlanderschap (65.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
      | B                                  | 310                   | 20010319                        |
      En de 'nationaliteit' is bewijzigd met de volgende gegevens
      | reden beëindigen (64.10) | datum ingang geldigheid (85.10) |
      | 410                      | 20190604                        |
      Als nationaliteithistorie wordt geraadpleegd met de volgende parameters
      | naam                | waarde                                                                 |
      | type                | RaadpleegMetPeriode                                                    |
      | burgerservicenummer | 555550001                                                              |
      | peildatum           | 2018-01-01                                                             |
      | fields              | nationaliteiten.redenOpname.code,nationaliteiten.datumIngangGeldigheid |
      Dan heeft de response de volgende 'nationaliteiten'
      | type                    | redenOpname.code | datumIngangGeldigheid.type | datumIngangGeldigheid.datum | datumIngangGeldigheid.langFormaat |
      | BehandeldAlsNederlander | 310              | Datum                      | 2001-03-19                  | 19 maart 2001                     |

Ik denk dat dit veel duidelijker beschrijft wat een beëindigde nationaliteit is/hoe een beëindigde nationaliteit tot stand is gekomen. En omdat het een betere functionele beschrijving is (geen impliciete aannames) kan deze scenario verhuizen van dev naar de functionele specs

Reply:
Ik snap ze allebei. Wat mij betreft geen voorkeur.Misschien een semantisch zeurvraagje, maar is de 'nationaliteit' gewijzigd met de volgende gegevens of is de 'nationaliteit' beeindigd met de volgende gegevens...

Reply:
@MelvLee ik heb ook situaties met meerdere wijzigingen (op verschillende momenten) op dezelfde nationaliteit. Wil je dan voor elke wijziging een aparte Gegeven stap? 

Ik ben het met je eens dat dit inderdaad duidelijker is. Wordt wel een lang verhaal bij meerdere nationaliteiten met elk wijzigingen.

```
      Gegeven de persoon met burgerservicenummer '555550001' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | reden beëindigen (64.10) | onjuist (84.10) | datum ingang geldigheid (85.10) |
      | 0131                  | 301                   |                          |                 | 20210604                        |
      |                       |                       | 404                      |                 | 20190516                        |
      | 0131                  | 301                   |                          |                 | 20010319                        |
      En de persoon met burgerservicenummer '555550001' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | reden beëindigen (64.10) | onjuist (84.10) | datum ingang geldigheid (85.10) |
      |                       |                       | 192                      |                 | 20210604                        |
      | 0001                  | 017                   |                          |                 | 20190516                        |
```

wordt dan:
```
      Gegeven de persoon met burgerservicenummer '555550001' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) |datum ingang geldigheid (85.10) |
      | 0131                  | 301                   |20010319                        |
      En de 'nationaliteit' is gewijzigd met de volgende gegevens
      | reden beëindigen (64.10) | datum ingang geldigheid (85.10) |
      | 404                      | 20190516                        |
      En de 'nationaliteit' is gewijzigd met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
      | 0131                  | 301                   | 20210604                        |
      En de persoon met burgerservicenummer '555550001' heeft een 'nationaliteit' met de volgende gegevens
      | nationaliteit (05.10) | reden opnemen (63.10) | datum ingang geldigheid (85.10) |
      | 0001                  | 017                   | 20190516                        |
      En de 'nationaliteit' is gewijzigd met de volgende gegevens
      | reden beëindigen (64.10) | onjuist (84.10) | datum ingang geldigheid (85.10) |
      | 192                      |                 | 20210604                        |
```

Reply:
de beschrijving zoals je voorstelt is m.i. duidelijker, maar bij meerdere wijzigingen wel minder overzichtelijk.

Reply:
Ik snap de onderste veel beter dan de bovenste. Ik ben inmiddels op de hoogte van de volgorde in de BRP, maar begrip van bovenste veronderstelt kennis van de BRP implementatie. De onderste is voor iedereen goed te begrijpen, of je nou BRP expert bent (RvIG) of niet. Daarnaast hoef je in de onderste geen gegevens te interpreteren. Ik heb daarom een hele sterke voorkeur voor de onderste.

Reply:
> Ik snap ze allebei. Wat mij betreft geen voorkeur.Misschien een semantisch zeurvraagje, maar is de 'nationaliteit' gewijzigd met de volgende gegevens of is de 'nationaliteit' beeindigd met de volgende gegevens...

Initieel zag de stap er ook als volgt uit: `En de 'nationaliteit' is beëindigd met de volgende gegevens`. Maar ik heb het veranderd naar gewijzigd (bewijzigd in dit geval), omdat ik zag dat in een ander scenario een beëindigd nationaliteit nog kan worden gewijzigd. Ik vond toen dat in dit geval wijzigen beter is dan beëindigen, maar het is zoals je aangeeft meer een semantische keuze.

Reply:
> de beschrijving zoals je voorstelt is m.i. duidelijker, maar bij meerdere wijzigingen wel minder overzichtelijk.

klopt, maar dat hangt ook af van de scenario. Wat wordt er met de scenario beschreven. Als het een edge case is, dan denk ik dat het makkelijker accepteren dat de scenario minder overzichtelijk is. Je zou ook een aparte gegeven stap kunnen definieren om een geval te specificeren, zodat het wel overzichtelijk wordt. Bijv.
```
Gegeven de persoon met burgerservicenummer '555550001' heeft een beëindigde 'nationaliteit' met datum ingang geldigheid '20190516' en datum tot '20210604'
```
Met deze stap wordt dan aangenomen dat het niet relevant is wat de nationaliteit, de opname reden en de reden van beëindigen zijn. Alleen de datums zijn relevant in de scenario